assets/js/collaborative-editor/components/ide/FullScreenIDE.tsx

```diff
@@ -343,6 +344,12 @@ export function FullScreenIDE({
 
   // Handle job selection from JobSelector
   const allJobs = workflow?.jobs || [];
+  const adjList = edgesToAdjList(workflow?.edges || []);
+  const ordinals = getJobOrdinals(adjList.list, adjList.trigger_id);
+  const sortedJobs = [...allJobs].sort((a, b) => {
+    return (ordinals[a.id] || Infinity) - ordinals[b.id] || Infinity;
```

Comment:
I think this part of the operation `ordinals[b.id] || Infinity` should be in parenthesis otherwise operator precedence may cause `|| Infinity` to apply to the entire subtraction result, not just  `ordinals[b.id]`. This would probably cause jobs disconnected from their trigger to not sort correctly.